pkg/oauthflow/interactive.go

```diff
@@ -59,17 +61,26 @@ func (i *InteractiveIDTokenGetter) GetIDToken(p *oidc.Provider, cfg oauth2.Confi
 	opts := append(pkce.AuthURLOpts(), oauth2.AccessTypeOnline, oidc.Nonce(nonce))
 	authCodeURL := cfg.AuthCodeURL(stateToken, opts...)
 	var code string
-	if err := open.Run(authCodeURL); err != nil {
-		// Swap to the out of band flow if we can't open the browser
-		fmt.Fprintf(os.Stderr, "error opening browser: %v\n", err)
-		code = doOobFlow(&cfg, stateToken, opts)
-	} else {
-		fmt.Fprintf(os.Stderr, "Your browser will now be opened to:\n%s\n", authCodeURL)
-		code, err = getCodeFromLocalServer(stateToken, redirectURL)
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "error getting code from local server: %v\n", err)
+	// Check if we are running integration tests
+	inTest := os.Getenv(integrationTest)
+	if inTest == "" {
```

Comment:
I think we can get around this by monkey patching `open.Run()` in the test which leaves the code as-is